internal/project/project.go

```diff
@@ -140,6 +140,27 @@ func (p *Project) GetSourceFile(fileName string, path tspath.Path, languageVersi
 	return nil
 }
 
+func (p *Project) GetImpliedNodeFormatForFileWorker(path string, packageJsonType string) core.ResolutionMode {
+	moduleResolution := p.GetProgram().Options().GetModuleResolutionKind()
```

Comment:
This code is identical to the one in compiler host, and only depends on options plus the arguments; it seems like this could be done in the parser when setting up the module, right? Then, this method doesn't need to exist at all.